### PR TITLE
feat: `CIBW_DEBUG_KEEP_CONTAINER`: debug container after build

### LINKED ISSUE #1320: Option to leave OCI container alive for debugging after build
### Description

When experimenting with the local setup, using `cibuildwheel --platform=linux`, for example, a container is created. The container is removed after the build finishes. If the build fails, there's no way to debug what caused it except for the logs. The roundtrip time of adding debug logs are pretty high for big software projects. For myself I've commented out these lines:

https://github.com/pypa/cibuildwheel/blob/f7669045f52f7ffc85ff6a3534b61ea90c00ad1b/cibuildwheel/oci_container.py#L148-L152

But I could PR a proper solution if someone talks me through the requirements, and if everyone agrees this is a good addition for debugging.

### Build log

_No response_

### CI config

_No response_

### PR DESCRIPTION
**Pull request by Helveg**
closes #1320. With this flag enabled, the containers of local (failing) builds can be debugged afterwards.

### CODE REVIEW

File: docs/options.md [hunk 1]
```diff
@@ -1426,6 +1426,38 @@ This option is not supported in the overrides section in `pyproject.toml`.
...
+    set CIBW_DEBUG_KEEP_CONTAINER=TRUE
+    ```
+
+!!! tab examples "MacOS"
+
+    ```shell
+    export CIBW_DEBUG_KEEP_CONTAINER=TRUE
+    ```
+
 ## Other
```
[the remark this answers is not in the document]
Comment (joerick) [reply docs/options.md:1461]: Yeah, I like the idea of putting CIBW_BUILD_VERBOSITY under debugging.

File: docs/options.md [hunk 2]
```diff
@@ -1426,6 +1426,38 @@ This option is not supported in the overrides section in `pyproject.toml`.
...
+option is provided for debugging purposes only.
+
+Default: Off (0).
+
+!!! caution
+    This option can only be set as environment variable on the host machine
+
+#### Examples
+
+!!! tab examples "Linux"
```
[the remark this answers is not in the document]
Comment (Helveg) [reply docs/options.md:1443]: I'm using it on Windows (PowerShell) with `$Env:CIBW_DEBUG_KEEP_CONTAINER = "TRUE"`, then I start the build with `cibuildwheel --platform linux`. I can reduce it to the standard `export CIBW_DEBUG_KEEP_CONTAINER=TRUE`

[the remark this answers is not in the document]
Comment (Helveg) [reply docs/options.md]: I added just a Linux example, people can figure out how to set env vars ^^

**Review by joerick — APPROVED** [review]
I'm kinda seeing options like this as tier-2 options, as they're for debugging purposes only, we don't need to be too strict around requiring tests etc.

**Comment by Helveg** [comment]
@joerick I also think the tests are merely failing because you've reached your CI quota with your CI provider?

> ⚠️ Failed to start an instance: FAILED_PRECONDITION: Monthly free compute limit exceeded!

**Pull request merged**